config/checker.go

```diff
@@ -2,45 +2,63 @@ package config
 
 import (
 	"fmt"
-	"github.com/shitpostingio/autopostingbot/config/structs"
 	"reflect"
+
+	"github.com/shitpostingio/autopostingbot/config/structs"
 )
 
 // checkMandatoryFields uses reflection to see if there are
 // mandatory fields with zero value.
+// When using with isReload flag, only checks for fields with reloadable:"true" tag
 func checkMandatoryFields(isReload bool, config structs.Config) error {
-	return checkStruct(isReload, reflect.TypeOf(config), reflect.ValueOf(config))
+	errors := ""
+	for _, err := range checkStruct(isReload, "root", reflect.TypeOf(config), reflect.ValueOf(config)) {
+		errors += fmt.Sprintln(err.Error())
+	}
+
+	if errors != "" {
+		return fmt.Errorf("\n%s", errors)
+	}
+
+	return nil
 }
 
 // checkStruct explores structures recursively and checks if
 // struct fields have a zero value.
-func checkStruct(isReload bool, typeToCheck reflect.Type, valueToCheck reflect.Value) (err error) {
-
+func checkStruct(isReload bool, parent string, typeToCheck reflect.Type, valueToCheck reflect.Value) (errors []error) {
+	errors = make([]error, 0)
 	for i := 0; i < typeToCheck.NumField(); i++ {
-
 		currentField := typeToCheck.Field(i)
 		currentValue := valueToCheck.Field(i)
-
 		switch currentField.Type.Kind() {
+		case
+			reflect.Uintptr,
+			reflect.UnsafePointer,
+			reflect.Ptr:
+			continue
+
 		case reflect.Struct:
-			err = checkStruct(isReload, currentField.Type, currentValue)
+			if errs := checkStruct(isReload, currentField.Name, currentField.Type, currentValue); errs != nil {
+				errors = append(errors, errs...)
+			}
+
 		case reflect.Slice:
-			err = checkSlice(isReload, currentField, currentValue)
-		default:
-			err = checkField(isReload, currentField, currentValue)
-		}
+			if errs := checkSlice(isReload, fmt.Sprintf("%s in %s", currentField.Name, parent), currentField, currentValue); errs != nil {
+				errors = append(errors, errs...)
+			}
 
-		if err != nil {
-			return
+		default:
+			if err := checkField(isReload, parent, currentField, currentValue); err != nil {
+				errors = append(errors, err)
+			}
 		}
-
 	}
 
-	return nil
+	return errors
 }
 
 // checkStruct explores slices recursively and checks they have a zero value.
```

Comment:
I've noticed the name of the function in the comment is wrong, could you change it as well, while you're at it?